spring-boot-starter-mesos/src/main/java/com/containersolutions/mesos/scheduler/CredentialFactory.java

```diff
@@ -0,0 +1,21 @@
+package com.containersolutions.mesos.scheduler;
+
+import com.containersolutions.mesos.scheduler.config.MesosConfigProperties;
+import com.google.protobuf.ByteString;
+import org.apache.mesos.Protos;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Creates credentials from user supplied principals and secrets
+ */
+public class CredentialFactory {
+    @Autowired
+    MesosConfigProperties mesosConfig;
+
+    public Protos.Credential create() {
+        return Protos.Credential.newBuilder()
+                .setPrincipal(mesosConfig.getPrincipal())
+                .setSecret(ByteString.copyFrom(mesosConfig.getSecret().getBytes()))
+                .build();
+    }
+}
```

Comment:
Potentially this could be replaced by

```java
@Bean
@ConditionalOnMissingBean
public Supplier<Protos.Credential> CredentialSupplier(MesosConfigProperties mesosConfdig) {
  return () -> Protos.Credential.newBuilder()
                  .setPrincipal(mesosConfig.getPrincipal())
                  .setSecret(ByteString.copyFrom(mesosConfig.getSecret().getBytes()))
                  .build();
}
```

in configuration